app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -59,6 +61,9 @@ def load_resource_from_solr
 
     # routed to /files/new
     def new
+      find_collections_with_edit_access
```

Comment:
OK, so this hooks into SelectsCollections. Can you pop this into a `before_action` hook?